### PR TITLE
Modified Nemotron 3 Nano recipe to use WSD LR schedule to match paper

### PR DESCRIPTION
**Pull request by OlegSudakov**
# What does this PR do ?

Modified Nemotron 3 Nano recipe to use WSD LR schedule. According to the paper:

We pretrained Nemotron 3 Nano 30B-A3B Base using the Warmup-Stable-Decay learning rate (LR) schedule for a total of 25 trillion tokens. We warmed up the LR over 8.4 billion tokens to a maximum of 10−3. We maintained the maximum LR for 80% of training (20 trillion tokens) and then finally decayed to a minimum of 10−5 during the last 20% of training (5 trillion tokens).

# Changelog

Modified src/megatron/bridge/recipes/nemotronh/nemotron_3_nano.py to use WSD instead of default cosine.

# GitHub Actions CI

See the [CI section](https://github.com/NVIDIA-NeMo/Megatron-Bridge/blob/main/CONTRIBUTING.md#-running-github-ci)in the Contributing doc for how to trigger the CI. A Nvidia developer will need to approve and trigger the CI for external contributors.

# Before your PR is "Ready for review"

**Pre checks**:

- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA-NeMo/Megatron-Bridge/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?

If you haven't finished some of the above items you can still open "Draft" PR.

# Additional Information

- Related to # (issue)

## Summary by CodeRabbit

* **Chores**
  * Updated scheduler configuration with new decay parameters to optimize training performance.

### CODE REVIEW

File: src/megatron/bridge/recipes/nemotronh/nemotron_3_nano.py [hunk 1]
```diff
@@ -154,11 +154,16 @@ def nemotron_3_nano_pretrain_config() -> ConfigContainer:
...
-    cfg.optimizer.lr = 1.6e-3
+    cfg.optimizer.lr = 1e-3
     cfg.optimizer.weight_decay = 0.1
-    cfg.scheduler.min_lr = 1.6e-5
-    cfg.scheduler.warmup_iters = 333
 
+    # Scheduler hyperparameters
+    cfg.scheduler.lr_decay_style = "WSD"
+    cfg.scheduler.lr_wsd_decay_style = "minus_sqrt"
+    cfg.scheduler.min_lr = 1e-5
```
Comment (liding-nv) [inline src/megatron/bridge/recipes/nemotronh/nemotron_3_nano.py]: `min_lr` is an optimizer param so i think it should be `cfg.optimizer.min_lr = 1.6e-5`

Comment (OlegSudakov) [reply src/megatron/bridge/recipes/nemotronh/nemotron_3_nano.py]: My bad, fixed

**Comment by liding-nv** [comment]
@OlegSudakov please fix the lint error and run all the CI tests. We can then merge after that. thanks
